modules/distributor/distributor.go

```diff
@@ -268,11 +268,11 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 		localCtx = user.InjectOrgID(localCtx, userID)
 
 		req := tempopb.PushBytesRequest{
-			Requests: make([][]byte, len(indexes)),
+			Requests: make([]tempopb.PreallocRequest, len(indexes)),
 		}
 
 		for i, j := range indexes {
-			req.Requests[i] = rawRequests[j][0:]
+			req.Requests[i].Request = rawRequests[j][0:]
```

Comment:
is `[0:]` necessary? I think it makes a new slice pointing at the same backing array. unsure if that's necessary.

Reply:
OK I didn't actually edit that array, but now that I think about it, I think `[0:]` is correct because it just copies over the slice, otherwise it would make a copy of the entire array

Reply:
Let me know if I'm missing something!

Reply:
I think that linked program will always be true because it copies the entire slice over. 

Check this one - https://play.golang.org/p/OFRbw97uZa3

I'm printing the address of the backing array after copying in two different ways

cc @mapno :)

Reply:
OK now I'm plain confused - https://play.golang.org/p/V1WGO_jh88Z

Reply:
There is a bit of trickery in that example since the initial variable is an array, rather than a slice. Although they look similar, they behave differently in this case. Arrays are values, so when copied, all the elements are copied over to the new variable. That is why we see two different pointers.

On the contrary, slices are pointers to where the elements are stored. Copying slices in go just copies the pointer, len and cap of the original slice. To actually duplicate the elements you need to use `copy()`. Unless I'm missing something, there is no difference between using `[0:]` or not (see https://play.golang.org/p/g2PAnWl-I2P).

As for the code, that line is copying the pointer of `rawRequests[j]` to `req.Requests[i].Request`. If `rawRequests[j]` is modified, it'll be shown in the _copy_. So it's important to know what happens to `rawRequests[j]` afterwards.

Reply:
Ah, thanks for explaining @mapno. That makes a lot of sense. I didn't realise that `[4]int{1, 2, 3, 4}` is the array representation.

In conclusion, in this case we should be ok using either, but since we were using `[0:]` earlier, I suggest we continue to use the same